jerry-core/ecma/operations/ecma-promise-object.h

```diff
@@ -58,6 +58,23 @@ typedef struct
   ecma_collection_header_t *reject_reactions; /**< list of PromiseRejectReactions */
 } ecma_promise_object_t;
 
+/**
+ * Use symbolic constant to represent the internal property name of
+ * promise related structures.
+ */
+typedef enum
+{
+  ECMA_PROMISE_PROPERTY_PROMISE, /**< [[Promise]] property */
+  ECMA_PROMISE_PROPERTY_RESOLVE, /**< [[Resolve]] property */
+  ECMA_PROMISE_PROPERTY_REJECT, /**< [[Reject]] property */
+  ECMA_PROMISE_PROPERTY_CAPABILITY, /**< [[Capability]] property */
+  ECMA_PROMISE_PROPERTY_HANDLER, /**< [[Handler]] property */
+  ECMA_PROMISE_PROPERTY_ALREADY_CALLED, /**< [[AlreadyCalled]] property */
+  ECMA_PROMISE_PROPERTY_INDEX, /**< [[Index]] property */
+  ECMA_PROMISE_PROPERTY_VALUE, /**< [[Values]] property */
+  ECMA_PROMISE_PROPERTY_REMAINING_ELEMENT /**< [[RemainingElement]] property */
```

Comment:
I am not sure what is "code above"? the code in your last comments or the code in the patch?

Do we have to assign the same integer number to multiple enums?